src/common/services/imageCache.service.js

```diff
@@ -0,0 +1,43 @@
+import angular from 'angular'
+import 'angular-environment'
+import retryService from './retry.service'
+
+const serviceName = 'imageCacheService'
+
+class ImageCache {
+  /* @ngInject */
+  constructor ($q, $http, envService, retryService) {
+    this.$q = $q
+    this.$http = $http
+    this.envService = envService
+    this.retryService = retryService
+    this.blobCache = new Map()
+  }
+
+  // Convert an image pathname into its absolute URL
+  makeAbsoluteUrl (url) {
+    return this.envService.read('imgDomainDesignation') + url
+  }
+
+  // Lookup an image pathname and return its blob URL if it is cached or its absolute URL otherwise
+  get (url) {
+    return this.blobCache.get(url) || this.makeAbsoluteUrl(url)
+  }
+
+  // Cache an image by its pathname and return a promise that resolves to its blob URL
+  cache (url) {
```

Comment:
The URL which is created is in the `location` header of the POST response for the image. It may be easier/faster to get it from there? Though, it won't have the image itself, which it looks like maybe we're getting from here.

Reply:
You're right that it's in the `Location` header. Here's where we extract that header to get the image URL. https://github.com/CruGlobal/give-web/blob/a61a9dd93819cf0fd77984c3f22abdfc1dbf945e/src/app/designationEditor/photoModal/photo.modal.js#L42 `imageCacheService` needs to cache all the variants of the image (cover, thumbnail, etc.), so just knowing the primary image URL isn't enough to start trying to load the cached images without reimplementing the naming logic on the client (i.e. `${imageUrl}.transform/GiveDesignationCoverPhoto/img.jpg`). We could try it that way, but I'm not sure that it will be faster because the slowdown is coming from getting lucky and requesting from a server that has replicated the image.